server/live_query/redis_live_query.go

```diff
@@ -101,14 +108,21 @@ func (r *redisLiveQuery) MigrateKeys() error {
 		return err
 	}
 
-	// identify which of those keys are in a deprecated format
```

Comment:
Note that `MigrateKeys` is called every time `fleet serve` is run, which was deemed fine when it was introduced, but now that we have enough keys that a SCAN can be a problem, we may want to run it only if a flag is explicitly set?

Reply:
Should we instead make it be something that we trigger through `fleetctl`?

Reply:
Yeah that would be a great place, I think.

Reply:
I'll add this subcommand in a separate PR.

Reply:
Or is it weird to have such a different approach to Redis migration vs mysql? E.g. this could be `fleet prepare redis`, or integrated in `fleet prepare db`?

Reply:
Hm, I like the idea of adding it to `prepare db`. @zwass any thoughts on extending prepare db to make sure we migrate redis things?

Reply:
This migration routine is designed to migrate in-progress live queries after the upgrade from before we supported Redis cluster to after? If that's the case I think we can consider removing it -- I think it's reasonable that in-progress live queries would stop after a large version bump.

Reply:
> This migration routine is designed to migrate in-progress live queries after the upgrade from before we supported Redis cluster to after

That's one part, but the part added by this PR is that it also migrates from using SCAN keys to find all live queries to saving the live query IDs in the active live queries set. That being said, your last part still make sense:

> I think it's reasonable that in-progress live queries would stop after a large version bump.

So in that case we could remove the redis keys migration altogether - so any existing live query would be stopped after the upgrade, given that it wouldn't be stored in the active queries set. Any pending ones would eventually expire (I believe the TTL is a week).

Reply:
I went ahead and removed the redis live query keys migration.